<!DOCTYPE html>
<html>
<head lang="en">
    <meta charset="UTF-8">
    <title>model-promise</title>
</head>
<style type="text/css">
    .divOne{
        background-color: #03b8cf;
    }
</style>
<script type="text/javascript" src="../../component/jq/jquery-1.4.4.min.js"></script>
<body>
<div id="one-div" class="divOne">model-promise</div>
<script type="text/javascript">
    var getUserInfo=new Promise(function(resolve,reject){
        $.ajax({
            type:"get",
            url:"/model/testPromise.do",
            success:function(data){
                var obj =eval('('+data+')');
                if(obj.Status=="2"){
                    console.log("success............+getUserInfo");
                    resolve(obj.ResultJson)//在异步操作成功时调用
                }else{
                    console.log("err............+getUserInfo");
                    reject(obj.ErrMsg);//在异步操作失败时调用
                }
            }
        });
    })

    var getDataList=new Promise(function(resolve,reject){
        $.ajax({
            type:"get",
            url:"/model/testPromise2.do",
            success:function(data){
                var obj =eval('('+data+')');
                if(obj.Status=="1"){
                    console.log("success............+getDataList");
                    resolve(obj.ResultJson)//在异步操作成功时调用
                }else{
                    console.log("err............+getDataList");
                    reject(obj.ErrMsg);//在异步操作失败时调用
                }
            }
        });
    })

//    getUserInfo.then(function(ResultJson){
//        //通过拿到的数据渲染页面
//        console.log("+++++++getUserInfo:"+ResultJson);
//    }).catch(function(ErrMsg){
//        //获取数据失败时的处理逻辑
//        console.log("+++++++getUserInfo:"+ResultJson);
//    })

    //Promise的all方法，等数组中的所有promise对象都完成执行
    Promise.all([getUserInfo,getDataList]).then(function([ResultJson1,ResultJson2]){
        //这里写等这两个ajax都成功返回数据才执行的业务逻辑
        console.log("all.....sucess................"+ResultJson1+'-------'+ResultJson2);
    }).catch(function(ErrMsg){
        //获取数据失败时的处理逻辑
        debugger;
        console.log("**********err:"+ErrMsg);

    })
</script>
</body>
</html>